main.go

```diff
@@ -31,15 +31,26 @@ func refresh(login string) {
 	expiry, found := getGcpExpiryFromCache(login)
 
 	if !found {
-		printDebug("login "+login+" not found in cache")
+		printDebug("login " + login + " not found in cache")
 		expiry = getGcpExpiryFromGcloud(login)
 		saveGcpExpiryToCache(login, expiry)
 	}
 
-	printDebug("expiry: "+expiry.String())
-	printDebug("now UTC: "+time.Now().UTC().String())
+	printDebug("expiry: " + expiry.String())
+	printDebug("now UTC: " + time.Now().UTC().String())
 
 	if time.Now().UTC().After(expiry) {
+		refreshGcpToken(login)
+		newExpiry := getGcpExpiryFromGcloud(login)
+
+		printDebug("new expiry: " + expiry.String())
```

Comment:
Did you mean to use the new expiry here?
```suggestion
		printDebug("new expiry: " + newExpiry.String())
```